moveit_setup_assistant/src/tools/collision_linear_model.cpp

```diff
@@ -129,8 +129,15 @@ QVariant CollisionLinearModel::data(const QModelIndex& index, int role) const
 
 DisabledReason CollisionLinearModel::reason(int row) const
 {
+  CollisionMatrixModel* m = qobject_cast<CollisionMatrixModel*>(sourceModel());
+  if (!m)
+  {
+    Q_ASSERT(m);
```

Comment:
In a release build, `Q_ASSERT` should be a no-op. Just silently continuing and returning an arbitrary value is actually hiding the error and will make it even harder to track the wrong behavior.